docs/_tooling/extensions/header_service/test/test_header_service.py

```diff
@@ -0,0 +1,321 @@
+    mock_extract_merge_commit_data.return_value = github_data
+    mock_generate_hash.return_value = hash_value
+    header_service = hs.HeaderService(
+        mock_app, "dummy-service", {"header_service_use_github_data": False}
+    )
+    assert header_service.request_from_directive(mock_directive) == [
+        need_data | github_data
+    ]
+    mock_extract_merge_commit_data.assert_called_once_with("file1.rst")
+
+
+@patch("header_service.header_service.HeaderService.request_from_directive")
+@patch("sphinx.application.Sphinx")
+def test_debug(mock_app, mock_request_from_directive):
+    debug_data = [{"key": "value"}]
+    mock_request_from_directive.return_value = debug_data
+    header_service = hs.HeaderService(mock_app, "dummy-service", None)
+    assert header_service.debug(None) == debug_data
+
+
+@patch("header_service.header_service.subprocess.run")
+def test_extract_merge_commit_data(run_mock):
+    lines = """abcdef
+John Doe
+Approved: {approver1} ( {approver1@mail.com} ) on {2024-11-1}
+Approved: {approver2} ( {approver2@mail.com} ) on {2024-11-2}
+Reviewed: {reviewer1} ( {reviewer1@mail.com} ) on {2024-12-1}
+Reviewed: {reviewer2} ( {reviewer2@mail.com} ) on {2024-12-2}
+Reviewed: {reviewer3} ( {reviewer3@mail.com} ) on {2024-12-3}"""
+
+    result_mock = MagicMock()
+    result_mock.returncode = 0
+    result_mock.stdout.decode.return_value = lines
+    run_mock.return_value = result_mock
+
+    assert hs._extract_merge_commit_data("/req/feature1") == {
+        "author": "John Doe",
+        "approver": ["approver1, approver1@mail.com", "approver2, approver2@mail.com"],
+        "reviewer": [
+            "reviewer1, reviewer1@mail.com",
+            "reviewer2, reviewer2@mail.com",
+            "reviewer3, reviewer3@mail.com",
+        ],
+        "hash": "abcdef",
+    }
+    run_mock.assert_called_once_with(
+        [
+            'git log --pretty="format:%H%n%an, %ae%n%b" --max-count=1 --merges --first-parent -p "/req/feature1'
+        ],
+        shell=True,
+        capture_output=True,
+    )
+
+
+@patch("header_service.header_service.subprocess.run")
+def test_extract_merge_commit_data_error(run_mock):
+    result_mock = MagicMock()
+    result_mock.returncode = 1
+    run_mock.return_value = result_mock
+
+    assert hs._extract_merge_commit_data("/req/feature1") == {
+        "author": "N/A",
+        "approver": "N/A",
+        "reviewer": "N/A",
+        "hash": "N/A",
+    }
+
+
+@patch("header_service.header_service._extract_approvers")
+@patch("header_service.header_service._extract_reviewers")
+@patch("header_service.header_service._extract_team_info")
+@patch("header_service.header_service._extract_org")
+@patch("header_service.header_service._extract_repo")
+@patch("header_service.header_service._extract_pull_request")
+@patch("header_service.header_service._extract_github_token")
+@patch("header_service.header_service.Auth.Token")
+@patch("header_service.header_service.Github")
```

Comment:
We need an abstraction class for all the github api calls, so we can have a single mock object here.